assets/src/edit-story/app/story/useStoryReducer/actions.js

```diff
@@ -0,0 +1,119 @@
+
+const addPage = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.ADD_PAGE, payload: { properties } } );
+
+const deletePage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageIndex } } );
+
+const deleteCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageIndex: null } } );
+
+const updatePageProperties = ( dispatch ) => ( { pageIndex, properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex, properties } } );
+
+const updateCurrentPageProperties = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex: null, properties } } );
+
+const duplicatePage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex } } );
+
+const duplicateCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex: null } } );
+
+const movePage = ( dispatch ) => ( { pageIndex, position } ) =>
+	dispatch( { type: types.MOVE_PAGE, payload: { pageIndex, position } } );
+
+const setCurrentPage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.SET_CURRENT_PAGE, payload: { pageIndex } } );
+
+const addElements = ( dispatch ) => ( { elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements } } );
+
+const addElement = ( dispatch ) => ( { element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements: [ element ] } } );
+
+const addElementsToPage = ( dispatch ) => ( { pageIndex, elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements } } );
+
+const addElementToPage = ( dispatch ) => ( { pageIndex, element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements: [ element ] } } );
+
+const deleteElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds } } );
+
+const deleteSelectedElements = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: null } } );
+
+const deleteElementById = ( dispatch ) => ( { elementId } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: [ elementId ] } } );
+
+const duplicateElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DUPLICATE_ELEMENTS, payload: { elementIds } } );
+
+const updateElementsById = ( dispatch ) => ( { elementIds, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds, properties } } );
+
+const updateElementById = ( dispatch ) => ( { elementId, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: [ elementId ], properties } } );
+
+const updateSelectedElements = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: null, properties } } );
+
+const moveElement = ( dispatch ) => ( { elementId, position } ) =>
```

Comment:
What does "moving" of element mean? Reordering in layer tree? Doesn't seem to be related to moving between pages here.

Reply:
Moving an element in this context is re-layering the element only. The name isn't great, I can see that (@miina asked about it too elsewhere). Is `reorderElement` better? Or maybe `arrangeElement`? (and similar for pages)

Reply:
I like "reorder".

Reply:
I just renamed them to `arrange` because it sounds nicer.

Looking at Keynote and Google Slides, `arrange` is used, but normally includes more than just reordering – also alignment, rotation, grouping, locking, etc.

Keynote | Google Slides
--- | ---
<img width="729" alt="Screen Shot 2019-12-11 at 6 16 30 PM" src="https://user-images.githubusercontent.com/637548/70668826-aa5f1a00-1c42-11ea-936b-af5487bca95e.png"> | <img width="845" alt="Screen Shot 2019-12-11 at 6 17 05 PM" src="https://user-images.githubusercontent.com/637548/70668827-aa5f1a00-1c42-11ea-8864-5a344685f1b3.png">

What does the people say? @spacedmonkey, @miina, @swissspidy, @pbakaus?

Should we name a function that moves a page's position in the page order as (the more correct but less elegant) `reorderPage` or (the cleaner but less specific) `arrangePage`?

Reply:
No strong opinion here. I think `arrange` is OK

Reply:
What about something like `movePageToPosition`? If `arrange` seems clear to most then that's OK with me, too.

Reply:
It would make sense to have the element reorder function named the same at the page one, and `moveElementToPosition` is pretty ambiguous.